clients/cobol-lsp-vscode-extension/src/services/reporter/TelemetryReporterImpl.ts

```diff
@@ -0,0 +1,113 @@
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import TelemetryReporter from "vscode-extension-telemetry";
+import {EXTENSION_ID, TELEMETRY_DEFAULT_CONTENT} from "../../constants";
+import {ExtensionUtils} from "../settings/util/ExtensionUtils";
+import {TelemetryEvent} from "./model/TelemetryEvent";
+import {TelemetryMeasurement} from "./model/TelemetryMeasurement";
+import {TelemetryReport} from "./TelemetryReport";
+
+export class TelemetryReporterImpl implements TelemetryReport {
+    public static getInstance(): TelemetryReporterImpl {
+        if (!TelemetryReporterImpl.instance) {
+            TelemetryReporterImpl.instance = new TelemetryReporterImpl(this.getTelemetryKeyId());
+        }
+        return TelemetryReporterImpl.instance;
+    }
+
+    private static instance: TelemetryReporterImpl;
+
+    /**
+     * This method return the value of the instrumentation key necessary to create the telemetry reporter from an
+     * external file configuration. If the file doesn't exists it returns a generic value that will not be valid
+     * for collect telemetry event.
+     */
+    private static getTelemetryKeyId(): string {
+        return fs.existsSync(this.getTelemetryResourcePath()) ? this.getInstrumentationKey() : TELEMETRY_DEFAULT_CONTENT;
+    }
+
+    private static getTelemetryResourcePath() {
+        return vscode.Uri.file(
+            path.join(ExtensionUtils.getExtensionPath(), "resources", "TELEMETRY_KEY")).fsPath;
+    }
+
+    private static getInstrumentationKey(): string {
+        return Buffer.from(fs.readFileSync(this.getTelemetryResourcePath(), "utf8").replace(/(\r\n|\n|\r)/gm, ""), "base64").toString();
+    }
+
+    private static covertData(content: TelemetryEvent) {
+        return {
+            categories: content.getCategories().toString(),
+            event: content.getEventName(),
+            IDE: ExtensionUtils.getIDEName(),
+            notes: content.getNotes(),
+            timestamp: content.getTimestamp(),
+            rootCause: content.getRootCause(),
+        };
+    }
+
+    private static convertMeasurements(content: Map<string, number>): TelemetryMeasurement {
+        if (!content) {
+            return undefined;
+        }
+
+        const result: TelemetryMeasurement = {};
+
+        for (const [key, value] of content) {
+            if (value) {
+                result[key] = value;
+            }
+        }
+        return result;
+    }
+
+    private readonly telemetryKeyId: string;
+    private reporter: TelemetryReporter;
+
+    private constructor(telemetryKeyId: string) {
```

Comment:
shortened form: we can remove above private fields and just define within constructor like **constructor(private** ...